FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1457 @@
+    static int lNumProcessed = 0;
+    bool xBreakOnNextProcessLoop = false;
+    bool xSubscribeProcessed = false;
+
+    while( 1 )
+    {
+        while( xQueueReceive( xCommandQueue, &xCommand, DEMO_TICKS_TO_WAIT ) )
+        {
+            pxCommand = &xCommand;
+
+            /* This demo requires the subscription to be present before the first publish. */
+            if( pxCommand->commandType == PUBLISH )
+            {
+                if( !xSubscribeProcessed )
+                {
+                    LogInfo( ( "Publish in queue before subscribe. Sending to back of queue." ) );
+                    prvAddCommandToQueue( pxCommand );
+                    continue;
+                }
+            }
+
+            xStatus = prvProcessCommand( pxCommand );
+
+            /* TODO: After reconnect implemented, add connect operation to front
+             * of queue if status was not successful. */
+            configASSERT( xStatus == MQTTSuccess );
+            lNumProcessed++;
+
+            if( pxCommand->commandType == PROCESSLOOP )
+            {
+                /* Add process loop back to end of queue. */
+                prvCreateCommand( PROCESSLOOP, NULL, NULL, &xNewCommand );
+                prvAddCommandToQueue( &xNewCommand );
+                lNumProcessed--;
+
+                if( xBreakOnNextProcessLoop )
+                {
+                    break;
+                }
+            }
+
+            /* Mark subscribed as being processed. */
+            if( pxCommand->commandType == SUBSCRIBE )
+            {
+                xSubscribeProcessed = true;
+            }
+
+            /* In this demo, exit after unsubscribing. */
+            if( pxCommand->commandType == UNSUBSCRIBE )
+            {
+                xBreakOnNextProcessLoop = true;
+            }
+        }
+
+        vTaskDelay( pdMS_TO_TICKS( 200 ) );
+
+        /* We have PUBLISH_COUNT publishes + 1 subscription */
+        if( lNumProcessed >= PUBLISH_COUNT + 1 )
+        {
+            break;
+        }
+    }
+
+    LogInfo( ( "Creating Disconnect operation" ) );
+    prvCreateCommand( DISCONNECT, NULL, NULL, &xNewCommand );
+    prvProcessCommand( &xNewCommand );
+    LogInfo( ( "Disconnected from broker" ) );
+}
+
+static void prvCommandCallback( CommandContext_t * pxContext )
+{
+    pxContext->complete = true;
+    xTaskNotify( pxContext->taskToNotify, pxContext->notificationBit, eSetBits );
+}
+
+/*-----------------------------------------------------------*/
+
+void prvPublishTask( void * pvParameters )
+{
+    ( void ) pvParameters;
+    Command_t xCommand;
+    MQTTPublishInfo_t xPublishInfo = { 0 };
+    MQTTPublishInfo_t pxPublishes[ PUBLISH_COUNT ];
+    char payloadBuf[ DEMO_BUFFER_SIZE ];
+    char topicBuf[ DEMO_BUFFER_SIZE ];
+    char * payloadBuffers[ PUBLISH_COUNT ];
+    char * topicBuffers[ PUBLISH_COUNT ];
+    CommandContext_t xContext;
+    CommandContext_t * pxContexts[ PUBLISH_COUNT ] = { 0 };
```

Comment:
I think only a size of PUBLISH_COUNT/2 needs to be allocated

Reply:
Yeah, the demo only used PUBLISH_COUNT - PUBLISH_COUNT/2 of these, which could be PUBLISH_COUNT/2 or PUBLISH_COUNT/2 + 1. Since pointers aren't really that large, I decided to just declare an array large enough to hold all of them, so that way the second `for` look wouldn't need to adjust all of its indices. Although, I could just have the asynchronous publishes be first and we wouldn't have to deal with those issues. What do you think?

Reply:
I see. I am ok with how it is.